src/scope.ml

```diff
@@ -208,12 +221,14 @@ let scope : mode -> sig_state -> env -> p_term -> tbox = fun md ss env t ->
     in
     aux env xs
   (* Scoping function for head terms. *)
-  and scope_head : env -> p_term -> tbox = fun env t ->
+  and scope_head : env -> p_term -> tbox = fun  env t ->
     match (t.elt, md) with
     | (P_Type          , M_LHS(_) ) ->
         fatal t.pos "[%a] is not allowed in a LHS." Print.pp Type
     | (P_Type          , _        ) -> _Type
-    | (P_Iden(qid,_)   , _        ) -> find_qid ss env qid
+    | (P_Iden(qid,_)   , _        ) ->
+      let allow_priv = match md with M_LHS(_) -> true | _ -> false in
```

Comment:
as the outer match is already on md (for its second component), it would be better to have two clauses in the outer matching:
```
| (P_Iden(qid,_)   , M_LHS(_)        ) -> find_qid true ss env qid
| (P_Iden(qid,_)   , _        ) -> find_qid false ss env qid (* as before *)
```